# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.h
# Commit: 0f43d0f3d542
# Full Hash: 0f43d0f3d542e22b92f4609576f050d77391a206
# Author: Kelsey Gilbert <jgilbert@mozilla.com>
# Date: 2021-12-15 21:51:13
# Description:
#   Bug 1745308 - Only warn once if ClientWaitSync cannot return availability. r=gfx-reviewers,nical
#   
#   This avoids spamming warnings across IPC that sometimes causes shmem
#   exhaustion.
#   
# ==============================================================================

diff -r 63b2fd522aa8 -r 0f43d0f3d542 dom/canvas/ClientWebGLContext.h
--- a/dom/canvas/ClientWebGLContext.h	Wed Dec 15 09:37:17 2021 +0000
+++ b/dom/canvas/ClientWebGLContext.h	Wed Dec 15 09:39:57 2021 +0000
@@ -485,6 +485,7 @@
   friend class webgl::AvailabilityRunnable;
 
   bool mCanBeAvailable = false;
+  bool mHasWarnedNotAvailable = false;
   bool mSignaled = false;
 
  public:
